source/main.c

```diff
@@ -367,67 +367,68 @@ int main()
     if (load_fat("sd2", &__io_gcsd2)) goto load;
 
 load:
-    if (dol)
+    if (!dol)
+    {
+        // If we reach here, all attempts to load a DOL failed
+        // Since we've disabled the Qoob, we wil reboot to the Nintendo IPL
+        delay_exit();
```

Comment:
You should probably add a log message here as well. "Rebooting into original IPL..."